pkg/services/query/query.go

```diff
@@ -130,6 +155,17 @@ func (s *Service) QueryData(ctx context.Context, user *user.SignedInUser, skipCa
 	return resp, nil
 }
 
+// buildErrorResponses applies the provided error to each query response in the list. These queries should all belong to the same datasource.
+func buildErrorResponses(err error, queries []*simplejson.Json) backend.Responses {
+	er := backend.Responses{}
+	for _, query := range queries {
+		er[query.Get("refId").MustString("A")] = backend.DataResponse{
```

Comment:
@marefr One thing I'm concerned about here is if the payload is for some reason missing a refId, and we end up overwriting an actual query with RefId = A with this error. Is this a valid concern? Should we do some sort of validation up front to ensure every query has a unique RefId?

Reply:
Yes, it's a valid concern I think, but could already happen with normal query processing so not a new problem since introducing support for querying multiple datasources.

I thought we had validation of this, but apparently not. Could be a breaking change enforcing this so would suggest opening a separate issue perhaps. WDYT?

Reply:
I will open a separate issue and ask around for ideas on what the resolution should be